### PR TITLE
Enable overlay component to have top margin

### PR DESCRIPTION
**Pull request by Lumenela**
Normally overlay component is centered vertically and horizontally, but in some scenarious we need to have more control over overlays positioning. I've added delegate method that returns top margin for the specified overlay component model.

### CODE REVIEW

**Comment by Lumenela** [comment]
@JohnSundell Agree, it's related to layout. But it's `HUBViewController` who positions overlay components, not `HUBComponentLayoutManager`. Is it okay to call `HUBComponentLayoutManager ` methods directly from `HUBViewController`?

**Comment by JohnSundell** [comment]
@Lumenela Sure, you can ask the component layout manager from the view controller 👍

File: include/HubFramework/HUBComponentLayoutManager.h [hunk 1]
```diff
@@ -112,6 +114,18 @@ typedef NS_ENUM(NSUInteger, HUBComponentLayoutContentEdge) {
...
+/**
+ *  Return top margin for overlay component with the specified model
+ *
+ *  @param componentModel The component model for the overlay component
+ *
+ *  @discussion If you want the overlay component to be centered top margin should be 0.
+ *  In case you want it to be shifted down, e.g. to reveal header component, return corresponding top margin
+ *  from this delegate method.
+ */
+- (CGFloat)topMarginForOverlayComponentWithModel:(id<HUBComponentModel>)componentModel;
```
Comment (JohnSundell) [inline include/HubFramework/HUBComponentLayoutManager.h]: We should expose the component model itself to the layout manager. That will quickly lead to very hard-coded messy rules around layout. Instead, we should use layout traits to make all layout decisions. This scales very well across a large code base, and enables more succinct rules to be defined. See the [Layout programming guide](https://spotify.github.io/HubFramework/layout-programming-guide.html) for more context. Feel free to introduce a new layout trait if you need to to accomplish your task 🙂

File: tests/mocks/HUBComponentLayoutManagerMock.h [hunk 2]
```diff
@@ -39,4 +39,6 @@
 /// Map of horizontal component offsets to use for an array of sets of layout traits
 @property (nonatomic, strong, readonly) NSMutableDictionary<NSArray<NSSet<HUBComponentLayoutTrait> *> *, NSNumber *> *horizontalComponentOffsetsForArrayOfLayoutTraits;
 
+@property (nonatomic, assign) CGFloat topMarginForOverlayComponent;
```
Comment (JohnSundell) [inline tests/mocks/HUBComponentLayoutManagerMock.h]: Document

**Comment by JohnSundell** [comment]
@Lumenela Any update on this? 😄

**Comment by Lumenela** [comment]
@JohnSundell It's just on pause :)
So layout manager should not access HUBComponentModel. Okay. Layout trait doesn't seem to be flexible enough. My component will always have the trait, but I want it to be enabled dynamic for certain cases.

**Pull request closed without merging**